### PR TITLE
gh-87092: Make jump target label equal to the offset of the target in the instructions sequence

### PR DESCRIPTION
**Pull request by iritkatriel**
In the instruction stream exposed by the compiler to Python for testing, the labels are no longer included as explicit entries. Instead, the offset of the target instruction in the stream is used. 

This is a step towards the codegen producing an instruction stream instead of basic blocks.

* Issue: gh-87092

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -9704,56 +9704,77 @@ instructions_to_cfg(PyObject *instructions, cfg_builder *g)
...
                 PyErr_SetString(PyExc_ValueError, "label out of range");
-                return ERROR;
+                goto error;
             }
-            jump_target_label lbl = {lbl_id};
+            is_target[oparg] = true;
+        }
+    }
+
+    for (Py_ssize_t i = 0; i < num_insts; i++) {
```
Comment (iritkatriel) [inline Python/compile.c]:
```suggestion
    for (int i = 0; i < num_insts; i++) {
```

**Comment by iritkatriel** [comment]
This windows test failure is unrelated.

Note that this PR is only changing test code.

File: Python/compile.c [hunk 2]
```diff
@@ -9704,56 +9704,77 @@ instructions_to_cfg(PyObject *instructions, cfg_builder *g)
 {
     assert(PyList_Check(instructions));
 
-    Py_ssize_t instr_size = PyList_GET_SIZE(instructions);
-    for (Py_ssize_t i = 0; i < instr_size; i++) {
+    Py_ssize_t num_insts = PyList_GET_SIZE(instructions);
+    bool *is_target = PyMem_Malloc(num_insts * sizeof(bool));
+    for (Py_ssize_t i = 0; i < num_insts; i++) {
+        is_target[i] = false;
+    }
```
Comment (erlend-aasland) [inline Python/compile.c]: These four lines (9708-9711), zeroing included, can be replaced with:

```c
    bool *is_target = PyMem_Calloc(num_insts, sizeof(bool));
```

File: Lib/test/support/bytecode_helper.py [hunk 3]
```diff
@@ -50,18 +50,13 @@ class CompilationStepTestCase(unittest.TestCase):
...
+    class Label:
+        pass
 
     def assertInstructionsMatch(self, actual_, expected_):
         # get two lists where each entry is a label or
-        # an instruction tuple. Compare them, while mapping
-        # each actual label to a corresponding expected label
-        # based on their locations.
+        # an instruction tuple. Normalize the labels to the
+        # instruction count of the target, adn compare the lists.
```
Comment (iritkatriel) [inline Lib/test/support/bytecode_helper.py]:
```suggestion
        # instruction count of the target, and compare the lists.
```

**Comment by iritkatriel** [comment]
@markshannon Did you say you've had look at this PR?

**Review by markshannon — APPROVED** [review]
Since this is only for testing, I'll trust you that is a worthwhile change.

It looks correct.

**Pull request merged**